test/patterns/gen-ai/aws-rag-appsync-stepfn-opensearch/aws-rag-appsync-stepfn-opensearch-serverless.test.ts

```diff
@@ -0,0 +1,198 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+import * as cdk from 'aws-cdk-lib';
+import { Template, Match } from 'aws-cdk-lib/assertions';
+import * as cognito from 'aws-cdk-lib/aws-cognito';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as oss from 'aws-cdk-lib/aws-opensearchserverless';
+import * as secret from 'aws-cdk-lib/aws-secretsmanager';
+import { RagAppsyncStepfnOpensearch, RagAppsyncStepfnOpensearchProps } from '../../../../src/patterns/gen-ai/aws-rag-appsync-stepfn-opensearch';
+
+
+describe('RAG Appsync Stepfn Open search construct', () => {
+
+  let ragTestTemplate: Template;
+  let ragTestConstruct: RagAppsyncStepfnOpensearch;
+  const cognitoPoolId = 'region_XXXXX';
+
+  afterAll(() => {
+    console.log('Test completed');
+  });
+
+  beforeAll(() => {
+    const ragTestStack = new cdk.Stack(undefined, undefined, {
+      env: { account: cdk.Aws.ACCOUNT_ID, region: cdk.Aws.REGION },
```

Comment:
Is it possible for this to be a snapshot instead from a mocked or fictitious account and region?

https://github.com/awslabs/generative-ai-cdk-constructs/blob/2a8188a3122c7c6c581d54fd74aab01e53f688f2/test/common/helpers/utils.test.ts#L31C5-L31C119